### PR TITLE
Schema parser: use full match instead of strings.HasPrefix

### PR DESCRIPTION
**Pull request by tuhao1020**
#1414 

https://github.com/gorilla/schema/issues/176

### CODE REVIEW

**Comment by tuhao1020** [comment]
@ReneWerner87 
Test_Cache_Expired failed:
```
=== RUN   Test_Cache_Expired
Error:     cache_test.go:77: Cache should not have expired: 1625478442551438654, 1625478442551768379
--- FAIL: Test_Cache_Expired (2.00s)
```

**Comment by ReneWerner87** [comment]
thank you, in the cache test there is a not stable test

if you have time you can analyze and stabilize it, i'll start the tests again for now

**Comment by ReneWerner87** [comment]
https://github.com/gorilla/schema/pull/178#issuecomment-873103020

i think he is right and you should include this in the solution

**Comment by tuhao1020** [comment]
@ReneWerner87 
I'm not very familiar with gorilla/schema, so I don't understand which case he was talking about,  can you provide a test case to me?

**Comment by ReneWerner87** [comment]
```go
	A struct {
		B int `schema:"b"`
	} `schema:"a,required"`
	
	C struct {
		D int `schema:"d,required"`
	} `schema:"c"`
```
I think he wants both to work

with your change you have destroyed the first case which checks that the object lying on "a" is not empty

in your example you only check if in "c" the property "d" is set


i.e. you would have to check if there is a dot in the current path and then do the partial check with it or if it is a path of the deepest element, then do the check for equality

i think the check with the HasPrefix is important, in my opinion there is still a dot missing in the string and as second condition the equality check

to understand the problem, i would recommend you to debug the unittests for this testcase, then you will see what is matched at this point and you should be able to look at the problem also from the second point of view

**Comment by tuhao1020** [comment]
@ReneWerner87  Thank you! I'll have a try

File: internal/schema/decoder.go [hunk 1]
```diff
@@ -160,7 +160,17 @@ func isEmptyFields(fields []fieldWithPrefix, src map[string][]string) bool {
 				// issue references:
 				// https://github.com/gofiber/fiber/issues/1414
 				// https://github.com/gorilla/schema/issues/176
-				if !isEmpty(f.typ, src[key]) && key == path {
+				nested := strings.Contains(key, ".")
```
Comment (ReneWerner87) [inline internal/schema/decoder.go:163]: use strings.IndexByte() ... it should be faster

Comment (tuhao1020) [reply internal/schema/decoder.go]: done

**Review by tuhao1020 — COMMENTED** [review]
viewed

**Review by tuhao1020 — COMMENTED** [review]
done

File: internal/schema/decoder.go (hunk 1 above)
Comment (ReneWerner87) [reply internal/schema/decoder.go]: thx

**Pull request merged**